var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -51,7 +51,12 @@ class PyTorch(PythonPackage, CudaPackage):
     ]
 
     version('master', branch='master', submodules=True)
-    version('1.4.0', tag='v1.4.0', submodules=True)
+    version('1.4.1', tag='v1.4.1', submodules=True)
+    # the following v1.4.0 tag is correct.
+    # v1.4.1 contains one fix on v1.4.0 to correct
+    # a broken tag on the fbgemm submodule
+    # see https://github.com/pytorch/pytorch/issues/35149
+    version('1.4.0', tag='v1.4.1', submodules=True)
```

Comment:
Thoughts on just removing 1.4.0 and only having 1.4.1? We can also do what we did with 0.4.1 and delete the fbgemm submodule since it's off by default anyway.

Oddly enough, even when I reproduce the steps in https://github.com/pytorch/pytorch/issues/35149, I still don't have an issue with fbgemm.

Reply:
This seems to be sensitive to the version of git in use. Using the system git (1.8.3.1) yields the issue, whereas using a newer version of git (> 2.19.1) does not. I'll probably be better able to answer your questions once I understand why that's happening.

Reply:
I think my preference would be for keeping 1.4.0 and 1.4.1, having them point to 1.4.0 and 1.4.1 respectively, deleting the submodule in 1.4.0, and adding a conflict for 1.4.0.

Reply:
You don't see this issue at all, correct? What version of git are you using with your tests?

Reply:
Correct, I'm using Apple Git 2.24.1

Reply:
AFAICT: upstream Git versions > 1.8.0 don't experience the issue because of https://github.com/git/git/commit/fb43e31f2b43076e7a30c9cd00d0241cb8cf97eb . FBGEMM had it's master branch force-pushed 11 days ago. You can see history from here: https://github.com/pytorch/FBGEMM/pull/316/commits

The newer upstream git versions try to do a ```git fetch origin 6aaaa4754fe2a7467c5f0584648ba47482caad7d``` and succeed at grabbing the submodule.

So from a comparing all of the different points:
```
6aaaa4754fe2a7467c5f0584648ba47482caad7d
399ea148f1403c100e6d601ec3587a621de96a84
and master
```
These commits seem to now be missing from the master branch and 399ea148f:
```
Update CMakeLists.txt to fix the OSS build issues (#195)
OSS build error fix (#193)
add SpMM generating code doesn't depend on shape of B (#194)
bug fix when handling remainder k block (#190)
porting Alek's sparse convolution using asmjit (#168)
minor changes in SpMM bench/test (#171)
code refactoring of FbgemmSpMM (#167)
fix build error in D18137995 (#158)
moving SpMM kernels from experimental to fbgemm (#151)
```
So I concur, simply renaming 1.4.1 is not equivalent to 1.4.0 - and we should probably proceed as you suggested. Tagging @jspark1105 to see if there are any other suggestions on how to potentially rescue 1.4.0 with fbgemm. (if those pull requests go away, does the 6aaaa4754 commit chain completely go away?)

Reply:
CC @yns88